src/components/ProductAddPage/ProductAdd.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from "react"
+import './ProductAdd.css'
+import { useHistory } from "react-router-dom";
+
+const ProductAdd = () => {
+  const history = useHistory();
+  const img = "https://previews.123rf.com/images/urfandadashov/urfandadashov1809/urfandadashov180901275/109135379-photo-not-available-vector-icon-isolated-on-transparent-background-photo-not-available-logo-concept.jpg"
+
+  const [errors, setErrors] = useState({
+    title: "",
+    img: "",
+    description: "",
+    rating: "",
+    gender: "",
+    category: "",
+    price: "",
+  })
+
+  const [product, setProduct] = useState({
+    title: "",
+    img: img,
+    description: "",
+    available: true,
+    rating: "",
+    gender: "",
+    category: "none",
+    price: 0,
+  })
+
+  const onCancel = () => {
+    history.push(`/`)
+  }
+
+  const onSave = () => {
+    const nextErrors = {
+      img: product.img.length === 0 ? 'Required' :
+        (/^https?:\/\/.*\.(jpeg|jpg|png)$/.test(product.img)) === false ? 'Invalid' : '',
+      rating: product.rating === '' ? 'Required' : '',
+      gender: product.gender === '' ? 'Required' : '',
+      category: product.category === 'none' ? 'Required' : '',
+      title: product.title === '' ? 'Required' : '',
+      description: product.description === '' ? 'Required' : '',
+      price: product.price === "" ? 'Required' :
+        /[a-zA-Z]/.test(product.price.toString()) ? "Enter Digits" : "",
+    }
+
+    setErrors(nextErrors)
+    const isValid = Object.values(nextErrors).every((value) => value === '')
+
+    if (!isValid) {
+      return
+    }
+
+    fetch("http://localhost:3001/products", {
+      method: 'POST',
+      headers: {
+        "Content-type": "application/json"
+      },
+      body: JSON.stringify(product)
+    }).then((response) => {
+      response.json().then(() => {
```

Comment:
No need to use then inside of then since they are chainable. So you can do next;

```
.then((response) => response.json()
.then(() => history.push("/"));
```